MekHQ/src/mekhq/campaign/personnel/enums/PersonnelRole.java

```diff
@@ -445,19 +605,35 @@ public static PersonnelRole parseFromString(final String text) {
                 default:
                     break;
             }
-        } catch (Exception ignored) {
+        } catch (Exception ignored) {}
 
+        // <50.1 compatibility
```

Comment:
The actual changes start here

Reply:
As I think about this (and other similar code), when was the conversion from Magic Numbers to Enum strings done? If pre-0.49.19, yank it.

Reply:
I'm fairly confident it was pre-49.19. I'm going to keep this in, though, because users updating from 50 are still going to run into the issue this addresses and for the sake of a few extra lines of code it maintains cross milestone compatibility. I absolutely agree with removing it once we hit our next compatibility threshold.